common/model-views.cpp

```diff
@@ -2917,6 +2921,102 @@ namespace rs2
+                                ir_stream->second.texture->try_pick(x, y, &ir_val);
+                                float pixel_ref = -1.0f;
+                                try 
+                                {
+                                    pixel_ref = viewer.ref.get_reflectivity(noise_est, max_usable_range, ir_val);
+                                }
+                                catch (...) {};
+                                std::string ref_str;
+                                if (pixel_ref == -1.0f)
+                                    ref_str = "N/A";
+                                else
+                                    ref_str = to_string() << std::dec  << round(pixel_ref * 100) << "%";
+
+                                if (!viewer.skip_option(RS2_OPTION_ENABLE_MAX_USABLE_RANGE))
+                                    ss << ", Reflectivity: "  << ref_str;
+                                else
+                                    ss << "\nReflectivity: " << ref_str;
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Draw IR reflectivity on IR frame
+            if (!viewer.skip_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY))
+            {
+                bool lf_exist = texture->get_last_frame();
+                if (lf_exist)
+                {
+                    auto ds = sensor_from_frame(texture->get_last_frame())->as<depth_sensor>();
+                    if (ds.supports(RS2_OPTION_ENABLE_IR_REFLECTIVITY) && (ds.get_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY) == 1.0f))
+                    {
+                        bool lf_exist = texture->get_last_frame();
+                        if (is_stream_alive() && texture->get_last_frame().get_profile().stream_type() == RS2_STREAM_INFRARED)
+                        {
+                            auto depth_stream = std::find_if(streams.cbegin(), streams.cend(), [](const std::pair<const int, stream_model> &stream)
+                            {
+                                return stream.second.profile.stream_type() == RS2_STREAM_DEPTH;
+                            });
+
+                            if (depth_stream != streams.end())
+                            {
+                                float depth_val = 0.0f;
+                                depth_stream->second.texture->try_pick(x, y, &depth_val);
+                                viewer.ref.add_input(depth_val, x, y);
+                                _reflectivity_data_collected = true;
+
+                                show_reflectivity = true;
+                                float noise_est = ds.get_option(RS2_OPTION_NOISE_ESTIMATION);
+                                auto mur_sensor = ds.as<max_usable_range_sensor>();
+                                auto max_usable_range = mur_sensor.get_max_usable_depth_range();
+                                auto ir_val = val;
+                                float pixel_ref = -1.0f;
```

Comment:
Better to just define `ref_str` here, default it to "N/A" and move line 3013 after line 3005, inside the try, no?

Does `get_reflectivity` ever return `-1`?

Reply:
get_reflectivity do not ever return -1.
But it can throw, so I needed an indication that the value was not set by it.
I will initialize it with "N/A"  and change the if/else to if

Reply:
I can change the function to return -1 when it can not calculate the value instead of throwing,
What do you think?